conduit/pipeline/pipeline.go

```diff
@@ -67,64 +61,60 @@ type Config struct {
 
 // Valid validates pipeline config
 func (cfg *Config) Valid() error {
-	if cfg.ConduitConfig == nil {
-		return fmt.Errorf("Config.Valid(): conduit configuration was nil")
+	if cfg.ConduitArgs == nil {
+		return fmt.Errorf("Args.Valid(): conduit args were nil")
 	}
-
-	if _, err := log.ParseLevel(cfg.PipelineLogLevel); err != nil {
-		return fmt.Errorf("Config.Valid(): pipeline log level (%s) was invalid: %w", cfg.PipelineLogLevel, err)
+	if cfg.PipelineLogLevel != "" {
+		if _, err := log.ParseLevel(cfg.PipelineLogLevel); err != nil {
+			return fmt.Errorf("Args.Valid(): pipeline log level (%s) was invalid: %w", cfg.PipelineLogLevel, err)
+		}
 	}
 
 	if len(cfg.Importer.Config) == 0 {
-		return fmt.Errorf("Config.Valid(): importer configuration was empty")
+		return fmt.Errorf("Args.Valid(): importer configuration was empty")
 	}
 
 	if len(cfg.Exporter.Config) == 0 {
-		return fmt.Errorf("Config.Valid(): exporter configuration was empty")
+		return fmt.Errorf("Args.Valid(): exporter configuration was empty")
 	}
 
 	return nil
 }
 
 // MakePipelineConfig creates a pipeline configuration
-func MakePipelineConfig(logger *log.Logger, cfg *conduit.Config) (*Config, error) {
-	if cfg == nil {
+func MakePipelineConfig(args *conduit.Args) (*Config, error) {
+	if args == nil {
 		return nil, fmt.Errorf("MakePipelineConfig(): empty conduit config")
 	}
 
-	// double check that it is valid
-	if err := cfg.Valid(); err != nil {
-		return nil, fmt.Errorf("MakePipelineConfig(): %w", err)
+	if !util.IsDir(args.ConduitDataDir) {
+		return nil, fmt.Errorf("MakePipelineConfig(): invalid data dir")
 	}
-	pCfg := Config{PipelineLogLevel: logger.Level.String(), ConduitConfig: cfg}
 
 	// Search for pipeline configuration in data directory
-	autoloadParamConfigPath := filepath.Join(cfg.ConduitDataDir, conduit.DefaultConfigName)
-
-	_, err := os.Stat(autoloadParamConfigPath)
-	paramConfigFound := err == nil
-
-	if !paramConfigFound {
-		return nil, fmt.Errorf("MakePipelineConfig(): could not find %s in data directory (%s)", conduit.DefaultConfigName, cfg.ConduitDataDir)
+	autoloadParamConfigPath, err := util.GetConfigFromDataDir(args.ConduitDataDir, conduit.DefaultConfigBaseName, []string{"yml", "yaml"})
+	if err != nil || autoloadParamConfigPath == "" {
+		return nil, fmt.Errorf("MakePipelineConfig(): could not find %s in data directory (%s)", conduit.DefaultConfigName, args.ConduitDataDir)
 	}
 
-	logger.Infof("Auto-loading Conduit Configuration: %s", autoloadParamConfigPath)
```

Comment:
Would this be useful to keep for debugging purposes?

Reply:
I think so, but the logger isn't initialized at this point (we don't know the log level until after processing the config file)